src/main/java/com/iexec/blockchain/tool/ChainConfig.java

```diff
@@ -16,32 +16,49 @@
 
 package com.iexec.blockchain.tool;
 
+import com.iexec.blockchain.tool.validation.ValidEthereumAddress;
 import lombok.AccessLevel;
 import lombok.Getter;
 
 import javax.annotation.PostConstruct;
 import javax.validation.ConstraintViolationException;
 import javax.validation.Validator;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
 import javax.validation.constraints.Positive;
 
+import lombok.ToString;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
 @Component
 @Getter
+@ToString
 public class ChainConfig {
 
     @Value("${chain.id}")
+    @Positive(message = "Chain id should be positive")
+    @NotNull
     private Integer chainId;
 
     @Value("${chain.node-address}")
+    @Pattern(message = "Node address should be a valid URL",
```

Comment:
How about `@URL`?
https://www.baeldung.com/hibernate-validator-constraints#3-url-and-html-validation
https://docs.jboss.org/hibernate/validator/6.2/api/org/hibernate/validator/constraints/URL.html

Reply:
So much cleaner, thanks!